### PR TITLE
Submit Lesson 5

### PR DESCRIPTION
**Pull request by codeeMadness**
Hi thay @quanghoang33,

Please kindly help to review my assignments.

Regards,
Hang Nguyen

### CODE REVIEW

File: Homework/Hang_Nguyen/lesson_5/insert-delete-getrandom-o1.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        self.map[val] = val
+        return True
+
+    def remove(self, val: int) -> bool:
+        if val in self.map:
+            self.map.pop(val)
+            return True
+        return False
+
+    def getRandom(self) -> int:
```
Comment (quanghoang33) [inline Homework/Hang_Nguyen/lesson_5/insert-delete-getrandom-o1.py:32]: can you make it O(1) ?

File: Homework/Hang_Nguyen/lesson_5/summary-about-ACID.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<h3>ACID stands for:</h3>
```
Comment (quanghoang33) [inline Homework/Hang_Nguyen/lesson_5/summary-about-ACID.md]: this is a good start. I suggest add more detail, recommend watching this video https://youtu.be/5ZjhNTM8XU8 if you don't like reading text book DDIA

**Review by quanghoang33 — CHANGES_REQUESTED** [review]

File: Homework/Hang_Nguyen/lesson_5/insert-delete-getrandom-o1.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# https://leetcode.com/problems/insert-delete-getrandom-o1/
+import random
+
+class RandomizedSet:
+
+    def __init__(self):
+        self.map = {}
+        
+
+    def insert(self, val: int) -> bool:
```
Comment (quanghoang33) [inline Homework/Hang_Nguyen/lesson_5/insert-delete-getrandom-o1.py:11]: O(1)

File: Homework/Hang_Nguyen/lesson_5/insert-delete-getrandom-o1.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        if val in self.map:
+            return False
+        self.map[val] = val
+        return True
+
+    def remove(self, val: int) -> bool:
```
Comment (quanghoang33) [inline Homework/Hang_Nguyen/lesson_5/insert-delete-getrandom-o1.py:19]: O(1)

**Review by quanghoang33 — CHANGES_REQUESTED** [review]

**Comment by codeeMadness** [comment]
Hi thay @quanghoang33 ,

Please kindly help to review my updates for assignment again.

Many thanks,
Hang Nguyen

File: Homework/Hang_Nguyen/lesson_5/summary-about-ACID.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
+<b><h3>A Transaction</h3></b>
+
+*A transaction is a way for an application to group several reads and writes together into a logical unit.*
```
Comment (quanghoang33) [inline Homework/Hang_Nguyen/lesson_5/summary-about-ACID.md:3]: can you give real-world examples for each bullet point? the most important point of a theory question is giving example

**Pull request merged**